### ISSUE TITLE
Rename application statuses to more descriptive and used words

### ISSUE THREAD

**Issue by brooksmtownsend** [wasmCloud/wadm#177]
Currently, our [application statuses](https://github.com/wasmCloud/wadm/blob/main/src/server/types.rs#L231) are as follows:
```rust
/// All possible status types
#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone, Copy, Default)]
#[serde(rename_all = "lowercase")]
pub enum StatusType {
    #[default]
    Undeployed,
    Compensating,
    Ready,
    Failed,
}
```

They cover a good variety of cases, but aren't necessarily the best naming conventions. The statuses don't inherently have anything wrong with them but ideally they reflect status types that orchestrators/control planes like [Nomad](https://www.nomadproject.io/) and [Kubernetes](https://kubernetes.io/) uses. I'd like to throw out a strawman and go from there 😄 

Here's what I would love to rename the statuses to:
1. Undeployed stays the same
2. Compensating -> Reconciling (To reflect when wadm has taken actions towards desired state and is awaiting events that would reach desired state)
3. Ready -> Deployed
4. Failed stays the same. Similar orchestrators reflect failed states as `pending` when there aren't enough applicable nodes, but I think our use here is fine.

The above assumes we like our current `put` and then `deploy` model, which I personally like because you can fearlessly put new versions of an application and then rollout at a future date.

### Additional Information
These are the available statuses and types in Nomad
- `Status`: The job's current state. It can have one of the following values:
  - `pending`: The job is currently waiting on scheduling.
  - `running`: The job has non-terminal allocations.
  - `dead`: All of the job's allocations and evaluations are terminal.
- `Type`: The type of job in terms of scheduling. It can have one of the following values:
  - `service`: Allocations are intended to remain alive.
  - `batch`: Allocations are intended to exit.
  - `system`: Each client gets an allocation.

Kubernetes follows a similar set of statuses for pods, and we don't have a notion of a "succeeded" state for wasmcloud apps.
Value | Description
-- | --
Pending | The Pod has been accepted by the Kubernetes cluster, but one or more of the containers has not been set up and made ready to run. This includes time a Pod spends waiting to be scheduled as well as the time spent downloading container images over the network.
Running | The Pod has been bound to a node, and all of the containers have been created. At least one container is still running, or is in the process of starting or restarting.
Succeeded | All containers in the Pod have terminated in success, and will not be restarted.
Failed | All containers in the Pod have terminated, and at least one container has terminated in failure. That is, the container either exited with non-zero status or was terminated by the system.
Unknown | For some reason the state of the Pod could not be obtained. This phase typically occurs due to an error in communicating with the node where the Pod should be running.

**Comment by lachieh** [comment]
TL;DR: 👍

Agreed on compensating, I like that word choice better. I initially thought that I didn't like the change to 'deployed' since the idea is to "deploy" an app and then wadm does what it can to get that application ready, regardless of status, but I can agree that it technically isn't fully deployed until it has launched everything it needs.

I'd like to also make the suggestion that we change `put` > `publish` in relation to uploading manifests.

**Comment by protochron** [comment]
I agree that `compensating` -> `reconciling` is the way to go. `pending` doesn't quite fit with how wadm thinks about things I think, because it's mostly issuing commands rather than waiting on containers/tasks to run.

**Comment by nebuk89** [comment]
I think it depends on whether we consider wadm to be acting like a pod manager or more like a whole deployment. You could look at the deployment types and go for something more like "progressing" for the replica sets. 

Or we could look at one of the popular cloud services, say ECS and pick up something like 'activating' (though this feels far more single service to me.) 

I am unsure if other K8s managers or production management tools give wider deployment status that reflects a super set of services and their state in other ways 🤔
